x-pack/plugins/infra/server/lib/alerting/inventory_metric_threshold/inventory_metric_threshold_executor.ts

```diff
@@ -191,6 +191,16 @@ export const createInventoryMetricThresholdExecutor = (libs: InfraBackendLibs) =
             : FIRED_ACTIONS.id;
 
         const alert = alertFactory(`${group}`, reason);
+        /**
```

Comment:
All the alert-related data that would be fetched from ES here (the executor function) is **one state behind**, always. Simply due to the execution order.  
The writing/indexing step is after the executor function is already executed (where this code is placed). I.e it requires the next execution of the executor function to read the value of the previous state.

I'm aware that we are talking about `ALERT_START` which is not going to be changed over the executions, but even though, we need at least 2 executions of the executor function to get the `ALERT_START`, because the first run we will get `null` (alert is not indexed yet), and the next execution to read it. But, and here is the point, if the alert got `recovered` after the first execution, the executor function will not run again, and we will be able to read the `ALERT_START`